lib/plugins/aws/package/compile/events/alb/lib/validate.js

```diff
@@ -55,6 +56,12 @@ module.exports = {
                 functionName
               );
             }
+            if (event.alb.authenticateCognito) {
+              albObj.authenticateCognito = this.validateCognitoAuth(event, functionName);
+            }
+            if (event.alb.authenticateOidc) {
```

Comment:
Can both authentication methods (`authenticateCognito` and `authenticateOidc`) be used for one event?

Reply:
Didn't [read anywhere](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-elasticloadbalancingv2-listener.html#cfn-elasticloadbalancingv2-listener-defaultactions) stating otherwise. I can limit it but I'm pretty sure you can have both

Reply:
As I read there can only be one. When we use `authenticateCognito`, then `Type` should be `authenticate-cognito`, and in case of `authenticateOidc`, `Type` should be set to `authenticate-oidc`

Reply:
Thats for each Listener Action. But you can have (numerous Actions per Listener)[https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-elasticloadbalancingv2-listener.html#cfn-elasticloadbalancingv2-listener-defaultactions]. The only requirement I noticed was that the last (with highest order) has to be of Type `forward`, `fixed-response` or `redirect`

Reply:
Indeed, Thanks for clarification!

Reply:
All good. I may have to hold off finishing this PR to the weekend. What are your release cycles like?

Reply:
We try to bring one minor release at least every two weeks, Next one will be either by end of this week or firts half of next